### PR TITLE
Support SDL_SetWindowTitle() by setting document.title.

### PR DESCRIPTION
**Pull request by dreamlayers**
DOSBox shows some information via the title, which is currently going nowhere. This modification will change the HTML page title.

I wonder if people hosting Emscripten generated code will see title changing as undesirable?

### CODE REVIEW

**Comment by juj** [comment]
I like this! A lot of pages change the window title, I don't see any reason why this wouldn't be appropriate here as well. What do you think @kripken ?

**Comment by kripken** [comment]
I suppose the argument could be made that there might be more than one app on a page; the "title" should be on top of the canvas, in a sense, but not necessarily the whole page. But, probably the majority of cases are going to be singleton canvases, so it would be ok...

The bigger question for me is, does this change anything right now? That is, if the user doesn't explicitly call a method to change the title, does SDL2 set it to "SDL2 Application" or something other default? If so this is a surprise we should avoid, otherwise, let's merge this.

**Comment by dreamlayers** [comment]
SDL 2 does not set a default window title. If the `title` parameter to [`SDL_CreateWindow()`](https://wiki.libsdl.org/SDL_CreateWindow) is NULL, then `SDL_SetWindowTitle()` is not called. The only other place it could be called is `SDL_RecreateWindow()`, where it is only called if `title` was set for the previous window. This is all in [src/video/SDL_video.c](https://github.com/emscripten-ports/SDL2/blob/master/src/video/SDL_video.c). Only other occurrences of `SetWindowTitle` are in declarations in header files and code in specific video drivers.

**Comment by dreamlayers** [comment]
What about adding a function to `Module` which is called to set the title? Then it could be set externally, overriding the default title-changing behaviour.

**Comment by jbaicoianu** [comment]
I'm actually working on a case which uses multiple SDL apps in one page now.  It would be nice to be able to get the title out of the app, but setting it directly on document.title might be a problem.  

Maybe if there's an optional callback you can set in module args when instantiating the app, `windowtitle: function(txt) { ... }`, or maybe the app could fire events that can be handled by the controlling page if custom behavior is needed?

**Comment by kripken** [comment]
I agree that an option on `Module` sounds like the best solution. Then multiple apps on one page would work. And we could make the default set the document title, of course, in the default HTML shell definition of Module.

If `Module.setWindowTitle` isn't defined, we'd either do nothing or warn, not sure which is best. Warn perhaps.

**Comment by dreamlayers** [comment]
What about setting `setWindowTitle` if it's undefined? So, if someone wants to override it, they can at any time, but if they do nothing it will be there and work. This would be done at some point during startup. I'm not sure what's the best place.
```JavaScript
if (typeof(Module['setWindowTitle']) == 'undefined')
    Module['setWindowTitle'] = function(title) { document.title = title };
````

**Comment by kripken** [comment]
Sounds reasonable. `shell.js` in the `ENVIRONMENT_IS_WEB` section, where we fill in other `Module` stuff as needed, is the right place.

The code here should then check and do nothing if it is not defined.

**Comment by dreamlayers** [comment]
I have created https://github.com/kripken/emscripten/pull/3192 and updated this pull request with code using `Module['setWindowTitle']()`.

File: src/video/emscripten/SDL_emscriptenvideo.c [hunk 1]
```diff
@@ -317,6 +318,16 @@ Emscripten_SetWindowFullscreen(_THIS, SDL_Window * window, SDL_VideoDisplay * di
     }
 }
 
+static void
+Emscripten_SetWindowTitle(_THIS, SDL_Window * window) {
+    EM_ASM_INT({
+      if (typeof Module['setWindowTitle'] != 'undefined') {
```
Comment (kripken) [inline src/video/emscripten/SDL_emscriptenvideo.c]: strict comparison here (`!==`)

**Pull request merged**